lib/services/platform-service.ts

```diff
@@ -137,12 +146,42 @@ export class PlatformService implements IPlatformService {
 				this.$fs.copyFile(newConfigFile, platformData.configurationFilePath).wait();
 			}
 
-			this.$projectDataService.initialize(this.$projectData.projectDir);
-			this.$projectDataService.setValue(platformData.frameworkPackageName, {version: installedVersion}).wait();
+			let frameworkPackageNameData: any = {version: installedVersion};
+			if(customTemplateOptions) {
+				frameworkPackageNameData.template = customTemplateOptions.selectedTemplate;
+			}
+			this.$projectDataService.setValue(platformData.frameworkPackageName, frameworkPackageNameData).wait();
 
 		}).future<void>()();
 	}
 
+	private getPathToPlatformTemplate(specifiedTemplate: string, frameworkPackageName: string): IFuture<any> {
+		return (() => {
+			if(!specifiedTemplate) {
+				// read data from package.json's nativescript key
+				// check the nativescript.tns-<platform>.template value
+				let nativescriptPlatformData = this.$projectDataService.getValue(frameworkPackageName).wait();
+				specifiedTemplate = nativescriptPlatformData && nativescriptPlatformData.template;
+			}
+
+			if(specifiedTemplate) {
+				let tempDir = temp.mkdirSync("platform-template");
+				try {
+					let installedTemplatePath = this.$npm.install(specifiedTemplate, tempDir, {"ignore-scripts": true, "production": true}).wait()[0][1];
+					return {
```

Comment:
If you rename installedTemplatePath, you can use the short syntax here. Not a stopper, though.